### PR TITLE
showing all release versions from git project

### PR DESCRIPTION
**Pull request by rrevanth**
Fixed grep problem that results in not showing all the releases of elixir.

### CODE REVIEW

**Comment by rrevanth** [comment]
I got over excited and made a new pull request.Sorry for the trouble but can you test this one.This one only shows the release versions.I don't understand why it's failing though!

File: bin/list-all [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
-# Fetch all tag names, and get only second column. Then remove all unnecesary characters.
-versions=$(curl -s $releases_path | grep "tag_name" | grep -oE '[0-9](\.[0-9]){2}' | tr '\n' ' ')
-echo $versions
+
+releases=$(curl -qs $releases_path | tr ',' '\n' | grep '"tag_name":')
+releases=${releases//\"tag_name\":/}
+releases=${releases//[\",]/}
+releases=${releases//v/}
+
+echo -e $releases
```
Comment (HashNuke) [inline bin/list-all]: Sort is required in the end to get latest first.
`echo -e $releases | sort`

Comment (HashNuke) [inline bin/list-all]: Actually, ignore my previous comment. This works fine. Sorting is already taken care of.

**Comment by HashNuke** [comment]
@rrevanth I'll need another pair of eyes on this. So I'll let @tuvistavie take a look once.

The previous (current) grep-line seems to omit everything with the older "vX.Y" pattern. They are older versions of elixir, we IMHO don't have to care about. @tuvistavie your call.

**Comment by tuvistavie** [comment]
@rrevanth Hi, thank you for the PR.
I forgot about `0.XX` versions when changing the `list-all`, sorry for that.
For the CI, I think there is an issue with the OSX build, I will take a look later.

I think it is a little complicated, can't we just keep the current code and change `'[0-9](\.[0-9]){2}'` into `'[0-9]+(\.[0-9]+){2}'`, which should give the same result?
Any case where you think the regex could fail?

**Comment by rrevanth** [comment]
@tuvistavie My code became complicated since I also put the branch listing in my previous pr and removed it on notice.I think regex is better than mine for releases

I think the updated regex fails at 1.3 .I suppose [0-9]+(\.[0-9]+){1,2} will cover everything.

If the versioning follows *.*.* as 1.3.0,then it wouldn't be a problem.

**Comment by tuvistavie** [comment]
> My code became complicated since I also put the branch listing in my previous pr and removed it on notice.I think regex is better than mine for releases

I see, let's go with the regexp then :smile: 

Elixir uses [semver](http://semver.org/), so there should be no version `1.3` but a version `1.3.0`, so I think we are safe assuming the version will be of the form `X.Y.Z`.

**Comment by HashNuke** [comment]
@rrevanth Sorry about holding off on this PR for a long time. Just noticed it. list-all versions have been fixed in another PR. So closing this PR.

Please do let us know if you find anything else though.

**Pull request closed without merging**